powerline/segments/vim/plugin/ctrlp.py

```diff
@@ -0,0 +1,102 @@
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+
+try:
+	import vim
+except ImportError:
+	vim = object()
+
+from powerline.bindings.vim import create_ruby_dpowerline
+
+
+def initialize():
+	global initialized
+	if initialized:
+		return
+	initialized = True
+	vim.command('echom 3210')
+	vim.command('let g:ctrlp_status = {}')
+	vim.command(
+		'''
+		execute 'function! Ctrlp_status_main(focus, byfname, regex, prev, item, next, marked)
+			let g:ctrlp_status = {\
+"focus": a:focus,\
+"byfname": a:byfname,\
+"regex": a:regex,\
+"prev": a:prev,\
+"item": a:item,\
+"next": a:next,\
+"marked": a:marked,\
+}
+			return ""
+		endfunction'
+		'''
+	)
+	vim.command(
+		'''
+		execute 'function! Ctrlp_status_prog(str)
+			let g:ctrlp_status["prog"] = a:str
+			return ""
+		endfunction'
+		'''
+	)
+	vim.command(
+		'''
+		let g:ctrlp_status_func = {\
+			'main': 'Ctrlp_status_main',\
+			'prog': 'Ctrlp_status_prog',\
+		}
+		'''
+	)
+
+initialized = False
+
+def marked(pl):
+	initialize()
+	status = vim.eval('g:ctrlp_status')
+	if 'prog' in status:
+		return []
+	return None if status['marked'] == ' <->' else [{
+		'highlight_groups': ['ctrlp:marked'],
+		'contents': status['marked'].strip()
```

Comment:
`vim_getvar` will most likely return `bytes` (`vim.eval` does this always in Python-2. In Python-3 `vim.eval` may return UnicodeDecodeError, though this is not really handled by `vim_getvar` (on old Vim versions this function still may return UnicodeDecodeError, but on newer it will always return `bytes` instance regardless of Python version and without any errors).)